include/glow/Runtime/HostManager/HostManager.h

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_RUNTIME_HOSTMANAGER_H
+#define GLOW_RUNTIME_HOSTMANAGER_H
+#include "glow/Graph/Graph.h"
+#include "glow/Runtime/Executor.h"
+#include "glow/Runtime/Partitioner.h"
+#include "glow/Runtime/Provisioner.h"
+
+#include <llvm/ADT/StringRef.h>
+
+#include <atomic>
+#include <unordered_map>
+namespace glow {
+class DeviceManager;
+class Context;
+enum ResultCode {
+  READY,
+  EXECUTED,
+  FAILED,
+  CANCELLED
+}; // This will likely be defined in one common runtime place.
+class HostManager final {
+  /// Count of current networks being run.
+  std::atomic<unsigned int> activeCount_;
+  /// Count of networks initialized on this node.
```

Comment:
"node" is kind of ambiguous here

Reply:
Also `totalCount_ ` is never decremented (because it's used to make `networkID`s) so it's the total number of networks ever seen on the host not the total number loaded right now but this is confusing because `activeCount_` sounds like it might be this however it is the number of in flight requests not networks (above it says "Count of current networks being run." but it could be multiple requests to the same network right so that should be changed?). I think it would be better to call these `activeRequestsCount_` (same thing with `activeRequestsLimit_`) and something like `nextNetworkId_` or something instead of `totalCount_ `